src/backend/crew_ai/llm_output_cleaner.py

```
@@ -156,184 +110,47 @@ def clean_action_input_lines(text: str) -> str:
-                removed_text = text[:start_pos]
-                logger.info(f"🧹 Removed duplicate/embedded code sections")
-                logger.debug(
-                    f"   Removed {len(removed_text)} chars before final code")
-                return cleaned_text
-
-        return text
+        # Pattern: "Action Input:" followed by extra text before the JSON
+        # e.g., "Action Input: Here's the input: {...}"
+        pattern = r'(Action Input:\s*)(?:Here\'?s?[^{]*|The input[^{]*|I\'ll use[^{]*)(\{)'
+        
+        def replacer(match):
```

Ruff (0.14.14)

Missing return type annotation for private function `replacer`

(ANN202)